blockchain/chaingen/generator.go

```diff
@@ -2148,12 +2180,14 @@ func (g *Generator) UpdateBlockState(oldBlockName string, oldBlockHash chainhash
 	// Remove existing entries.
 	wonTickets := g.wonTickets[oldBlockHash]
 	delete(g.blocks, oldBlockHash)
+	delete(g.blockHeights, oldBlockHash)
 	delete(g.blocksByName, oldBlockName)
 	delete(g.wonTickets, oldBlockHash)
 
 	// Add new entries.
 	newBlockHash := newBlock.BlockHash()
 	g.blocks[newBlockHash] = newBlock
+	g.blockHeights[newBlockHash] = newBlock.Header.Height
```

Comment:
The existing height in should be reinsert for the new hash, not the now potentially invalid one.

```Go
wonTickets := g.wonTickets[oldBlockHash]
existingHeight := g.blockHeight(oldBlockHash)
...
g.blockHeights[newBlockHash] = existingHeight
```